rust/agama-dbus-server/src/network/model.rs

```diff
@@ -480,6 +550,47 @@ pub struct LoopbackConnection {
 }
 
 #[derive(Debug, Default, PartialEq, Clone)]
+pub struct BondConnection {
+    pub base: BaseConnection,
+    pub bond: BondConfig,
+}
+
+#[derive(Debug, Default, Clone, PartialEq)]
+pub struct BondOptions(pub HashMap<String, String>);
+
+impl TryFrom<String> for BondOptions {
```

Comment:
Perhaps it would be better to implement [FromStr](https://doc.rust-lang.org/std/str/trait.FromStr.html) instead. Does it fit your use case? IMHO it is more versatile and you do not need to consume the `String`.